pandas/tests/internals/test_internals.py

```diff
@@ -350,7 +358,10 @@ def test_duplicate_ref_loc_failure(self):
         blocks[1].mgr_locs = np.array([0])
 
         # test trying to create block manager with overlapping ref locs
-        with pytest.raises(AssertionError):
+
+        msg = "Gaps in blk ref_locs"
```

Comment:
I don't think we should test this, this is an internal error message

(and it's also not a very good error message)

Reply:
> (and it's also not a very good error message)

Isn't that exactly the reason why we should test them in the first place?  Puts more light on the quality of the error messages that we raise, internal or not.

Reply:
> Isn't that exactly the reason why we should test them in the first place

If we, in one go, improve the error message. Yes, that is fine, but otherwise I don't think it is worth the effort.

Reply:
> Yes, that is fine, but otherwise I don't think it is worth the effort.

Debatable.  If you're someone who has just started working on this codebase, I wouldn't want them to also modify the error message.  You can easily spend another PR figuring out what the proper error message is.

Reply:
I agree with @jorisvandenbossche here I don't think worth testing against error messages for `AssertionError`, since that is purely internal

Reply:
> since that is purely internal

I still don't find that to be valid justification.  Internal or not, we use error messages for a reason: because they provide some utility in development.  If these error messages (no the `Exception` class itself) had no use because they're internal, we should just get rid of them.

Furthermore, error messages serve to differentiate between which error is raised.  The `managers.py` file is littered with `AssertionError` lines.  It would be good to know that the correct one is being raised.

Reply:
I am actually ok with testing these asserts; they are the current behavior. If things change then this test should fail. That said I wouldn; spend too much effort on this.

Reply:
> because they provide some utility in development.

Yes, they are useful for that. But also because they are only used in development, they can also easily change when refactoring some internals (without that there are user-facing consequences).

But every test (and certainly a test with a match) takes also effort to write, review and maintain (eg update if you update something internally). It's a trade-off. So I am not saying those internal error messages are not useful at all, but I still think they are not worth the effort to properly test with full coverage.